spring-cloud-azure-appconfiguration-config-web/src/main/java/com/microsoft/azure/spring/cloud/config/web/AppConfigurationWebAutoConfiguration.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config.web;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.microsoft.azure.spring.cloud.config.AppConfigurationRefresh;
+
+@Configuration
+public class AppConfigurationWebAutoConfiguration {
+
+    @Configuration
+    @ConditionalOnClass(RefreshEndpoint.class)
+    static class AppConfigurationWatchAutoConfiguration {
+
+        @Bean
+        public ConfigListener configListener(AppConfigurationRefresh appConfigurationRefresh) {
```

Comment:
If the configListener bean depends on RefreshEndpoint.class, should we put @Component annotation on the ConfigListner class?

Reply:
@saragluna ConfigListener is already marked as a component. See line 16 of the ConfigListener.

Reply:
Sorry for not making myself clear. I mean we tend to configure this bean on some condition, however, putting a @component annotation will always initiate that bean.  Is this what we want?

Reply:
@saragluna Yes, at least for now. if you look at the package as a whole it only has two classes right now, which are this configuration class and the listener. This might not always be the case but at least for now the listener should always be created when using this package.